catalog/src/test/java/it/eng/catalog/rest/protocol/CatalogProtocolControllerTest.java

```diff
@@ -0,0 +1,167 @@
+package it.eng.catalog.rest.protocol;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+
+import it.eng.catalog.model.Catalog;
+import it.eng.catalog.model.CatalogError;
+import it.eng.catalog.service.CatalogService;
+import it.eng.catalog.service.DatasetService;
+import it.eng.catalog.util.MockObjectUtil;
+import it.eng.tools.model.DSpaceConstants;
+
+@WebMvcTest(controllers = {CatalogProtocolController.class})
```

Comment:
This leans more to integration testing, maybe we should move this code, without mocking service (when applicable) to integration part?

Reply:
Probably will be done so as soon as we have MongoDB code merged and testcontainers library added